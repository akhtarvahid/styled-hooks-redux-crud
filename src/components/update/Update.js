import React from "react";
import { Modal, Button, Form } from "react-bootstrap";
import './update.scss';
import { UPDATE_ITEM } from '../../utils/constant';

function Update({ status, handleClose, handleChange,
  submitUpdateForm, state }) {
  const { title, description } = state;
  return (
      <Modal show={status} onHide={handleClose} animation={true} centered>
        <Modal.Header closeButton className='modal-content_update'>
          <Modal.Title>{UPDATE_ITEM}</Modal.Title>
        </Modal.Header>
        <Modal.Body>
          <Form onSubmit={submitUpdateForm}>
            <Form.Group controlId="formBasicTitle">
              <Form.Label>Title</Form.Label>
              <Form.Control
                required
                type="text"
                placeholder="Enter Title"
                name="title"
                value={title || ''}
                onChange={handleChange}
              />
            </Form.Group>

            <Form.Group controlId="formBasicDescription">
              <Form.Label>Description</Form.Label>
              <Form.Control
                required
                type="text"
                placeholder="Enter Description"
                name="description"
                value={description || ''}
                onChange={handleChange}
              />
            </Form.Group>
          </Form>
        </Modal.Body>
        <Modal.Footer>
          <Button variant="danger" onClick={handleClose}>
            Close
          </Button>
          <Button variant="success" onClick={submitUpdateForm}>
            Save Changes
          </Button>
        </Modal.Footer>
      </Modal>
  );
}

export default Update;