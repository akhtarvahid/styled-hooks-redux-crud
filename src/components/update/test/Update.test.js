import React from 'react';
import renderer from 'react-test-renderer';
import Update from '../Update';

test('should render Update Component', () => {
    let fakeProps = {
        id: 1,
        title: "title",
        description: "description",
        completed: false
    };
    const tree = renderer
        .create(<Update state={fakeProps} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
