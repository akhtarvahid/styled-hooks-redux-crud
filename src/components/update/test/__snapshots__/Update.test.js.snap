// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Update Component 1`] = `null`;
